tests/PHPStan/Analyser/data/bug-4371.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bug4371;
+
+use function PHPStan\Testing\assertType;
+use Exception;
+
+class HelloWorld
+{
+	/**
+	 * @param Exception $exception
+	 * @param class-string $classString
+	 */
+	public function sayHello(Exception $exception, $classString): void
+	{
+		assertType('bool', is_a($exception, $classString));
+		assertType('bool', is_a($exception, $classString, true));
+		assertType('bool', is_a($exception, $classString, false));
+
+		$exceptionClass = get_class($exception);
+		assertType('false', is_a($exceptionClass, $classString));
+		assertType('bool', is_a($exceptionClass, $classString, true));
+		assertType('false', is_a($exceptionClass, $classString, false));
```

Comment:
with https://github.com/phpstan/phpstan-src/pull/1306/commits/3e955aff9cad10bc653e16fb727a24b69bd655e0 I dropped the conditional return type stub again, as I am no longer sure its worth to report about `string` type errors for $class.. might be tedious.

since we already have extensions for `is_a` in place the stub itself was nevertheless only partly usefull, I think.